python/tk_hiero_export/sg_shot_processor.py

```diff
@@ -246,35 +306,320 @@ def startProcessing(self, exportItems):
+        collateShotNames = properties.get('collateShotNames', False)
+
+        return (collateTracks, collateShotNames)
+
+    def _getCutData(self, hiero_sequence):
+        """
+        Returns a dict of cut data for the supplied hiero sequence.
+
+        :param hiero_sequence: `hiero.core.Sequence` object
+        :return: dict - cut data fields
+        """
+
+        parent_entity = None
+
+        try:
+            # get the parent entity in SG that corresponds to the hiero sequence
+            parent_entity = self.app.execute_hook_method(
+                "hook_get_shot",
+                "get_shot_parent",
+                hiero_sequence=hiero_sequence,
+                data=self.app.preprocess_data,
+                upload_thumbnail=False,
+            )
+        except TankHookMethodDoesNotExistError, e:
+            # the method doesen't exist in the hook. the hook may have been
+            # overridden previously and not updated to include this method.
+            # this will imply a Cut stream without a parent entity. For now,
+            # we will log a warning and continue.
+            self.app.log_warning(
+                "The method 'get_shot_parent' could not be found in the "
+                "'hook_get_shot' hook. In order to properly link the "
+                "Cut entity in SG, you will need to implement this method "
+                "to return a Sequence, Episode, or some other entity "
+                "that corresponds to the Hiero sequence in your workflow."
+            )
+            pass
+
+        # determine which revision number of the cut to create. look for an
+        # existing Cut with the sequence name with the same parent.
+        sg = self.app.shotgun
+        prev_cut = sg.find_one(
+            "Cut",
+            [["code", "is", hiero_sequence.name()],
+             ["entity", "is", parent_entity]],
+            ["revision_number"],
+            [{"field_name": "revision_number", "direction": "desc"}]
+        )
+
+        if prev_cut is None:
+            # no matching Cut, start out at version 1
+            next_revision_number = 1
+        else:
+            # match! use the next revision number
+            next_revision_number = prev_cut["revision_number"] + 1
+
+        self._app.log_debug(
+            "The cut revision number will be %s." % (next_revision_number,))
+
+        # retrieve the cut type from the processor presets
+        properties = self._preset.properties().get(
+            'shotgunShotCreateProperties', {})
+        cut_type = properties.get("sg_cut_type", "")
+
+        # the bulk of the cut data. the rest will be populated as the individual
+        # shots are processed in the cut
+        return  {
+            "project": self.app.context.project,
+            "entity": parent_entity,
+            "code": hiero_sequence.name(),
+            "sg_cut_type": cut_type,
+            "description": "Automatically created by the Hiero Shot exporter.",
```

Comment:
If you are using Hiero/Nuke Studio you'll probably end up with a lot of Cuts that have this in the description, which doesn't really look great and doesn't add any info.  I'd say leave it empty.